connectors/sources/sharepoint_online.py

```diff
@@ -718,16 +711,23 @@ async def sites(
 
     async def _all_sites(self, sharepoint_host, allowed_root_sites):
         select = ""
-        async for page in self._graph_api_client.scroll(
-            f"{GRAPH_API_URL}/sites/{sharepoint_host}/sites?search=*&$select={select}"
-        ):
-            for site in page:
-                # Filter out site collections that are not needed
-                if [WILDCARD] != allowed_root_sites and site[
-                    "name"
-                ] not in allowed_root_sites:
-                    continue
-                yield site
+        try:
+            async for page in self._graph_api_client.scroll(
+                f"{GRAPH_API_URL}/sites/{sharepoint_host}/sites?search=*&$select={select}"
+            ):
+                for site in page:
+                    # Filter out site collections that are not needed
+                    if [WILDCARD] != allowed_root_sites and site[
+                        "name"
+                    ] not in allowed_root_sites:
+                        continue
+                    yield site
+        except PermissionsMissing as exception:
+            if allowed_root_sites == [WILDCARD]:
+                msg = "The configuration field 'Comma-separated list of sites' with '*' value is only compatible with 'Sites.Read.All' permission."
+            else:
+                msg = "To enumerate all sites, the connector requires 'Sites.Read.All' permission"
```

Comment:
how would we end up here, if `allowed_root_sites != [WILDCARD]`?

Reply:
That will fall under this block of code https://github.com/elastic/connectors/blob/efaa2d48b281110004d765390b8d9d218866b7e5/connectors/sources/sharepoint_online.py#L695-L710

`_all_sites` method is only responsible for this `if` condition - https://github.com/elastic/connectors/blob/efaa2d48b281110004d765390b8d9d218866b7e5/connectors/sources/sharepoint_online.py#L691

Reply:
`or enumerate_all_sites:`

That was the piece I was missing. Thanks!